src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationService.java

```diff
@@ -0,0 +1,393 @@
+     */
+    public Page<User> searchMembersOfConversation(Course course, Conversation conversation, Pageable pageable, String searchTerm,
+            Optional<ConversationMemberSearchFilters> filter) {
+        if (filter.isEmpty()) {
+            return userRepository.searchAllByLoginOrNameInConversation(pageable, searchTerm, conversation.getId());
+        }
+        else {
+            switch (filter.get()) {
+                case INSTRUCTOR -> {
+                    return userRepository.searchAllByLoginOrNameInConversationWithCourseGroup(pageable, searchTerm, conversation.getId(), course.getInstructorGroupName());
+                }
+                case TUTOR -> {
+                    // searches for both tutors and editors
+                    return userRepository.searchAllByLoginOrNameInConversationWithEitherCourseGroup(pageable, searchTerm, conversation.getId(), course.getEditorGroupName(),
+                            course.getTeachingAssistantGroupName());
+                }
+                case STUDENT -> {
+                    return userRepository.searchAllByLoginOrNameInConversationWithCourseGroup(pageable, searchTerm, conversation.getId(), course.getStudentGroupName());
+                }
+                case CHANNEL_MODERATOR -> {
+                    assert conversation instanceof Channel : "The filter CHANNEL_MODERATOR is only allowed for channels!";
+                    return userRepository.searchChannelModeratorsByLoginOrNameInConversation(pageable, searchTerm, conversation.getId());
+                }
+                default -> throw new IllegalArgumentException("The filter is not supported.");
+            }
+        }
+
+    }
+
+    /**
+     * Switch the favorite status of a conversation for a user
+     *
+     * @param conversationId the id of the conversation
+     * @param requestingUser the user that wants to switch the favorite status
+     * @param isFavorite     the new favorite status
+     */
+    public void switchFavoriteStatus(Long conversationId, User requestingUser, Boolean isFavorite) {
+        var participation = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserId(conversationId, requestingUser.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Conversation participant not found!"));
+        participation.setIsFavorite(isFavorite);
+        conversationParticipantRepository.save(participation);
+    }
+
+    /**
+     * Switch the hidden status of a conversation for a user
+     *
+     * @param conversationId the id of the conversation
+     * @param requestingUser the user that wants to switch the hidden status
+     * @param hiddenStatus   the new hidden status
+     */
+    public void switchHiddenStatus(Long conversationId, User requestingUser, Boolean hiddenStatus) {
+        var participation = conversationParticipantRepository.findConversationParticipantByConversationIdAndUserId(conversationId, requestingUser.getId())
```

Comment:
The method for `orElseThrow` should be in the Repository.

Reply:
Done in 1db0a029fc229bb2463a3fae195117c3a1f3aa81